### DISCUSSION TITLE
Remove SemVer as a requirement for verified mods

### DISCUSSION THREAD

**Discussion by EladNLG** [R2Northstar/NorthstarLauncher#693] [General]
I think this one is due some explanation.

SemVer is a versioning system that is good on paper, but is unused through the entirety of northstar... except verified mods.

Whilst the argument of "it makes it easier to separate breaking and not-breaking patches" is true... _the vast majority of changes to both maps and gamemodes would break compatibility with previous versions._ This means 2 numbers in the semver are effectively useless (as they will never get incremented 99% of the time)

Especially when most users consider MAJOR increments to be big updates, you have to resort to some system such as `YYYYMMDD.0.0` to avoid confusion and ambiguity about the size of the updates.

## My Proposal

Consider every change breaking. Servers and clients should be encouraged to update to the newest version once it releases, regardless of whether its breaking or not.

**Comment by GeckoEidechse** [comment]
In regards to verified mods, getting the client to update is trivial. In fact it already pulls the update, even on newer releases.

So the questions is basically, should we allow client with newer patch release of a mod join a server that might be running an older patch? Or should we force the client to download the older patch if the server is slow to update?

**Reply by Alystrasz** [reply]
Could you ELI5 pls? :)
(I kinda agree with Elad here, full answer to come later)

**Reply by EladNLG** [reply]
Force old version - because most new versions are breaking changes (and we do not know if there is backwards-compatibility between versions), imo we should go the safe route and always download the same version as the server.

**Reply by Alystrasz** [reply]
That's the way I initially thought of semver with mods, patch releases should be compatible (*e.g.* client with `v0.2.5` should be able to join a `v0.2.4` server); this however doesn't apply the other way according to semver, and since mods have a unique version number for client and server parts, that's another point for removing this criterion.
